logstash-core/lib/logstash/pipeline.rb

```diff
@@ -176,8 +179,10 @@ def stopped?
   end
 
   def start_workers
-    @inflight_batches = {}
-
+    @filter_queue_client.set_events_metric(metric.namespace([:stats, :events]))
```

Comment:
These methods should probably be `@filter_queue_client.events_metric = ...` to be more ruby-like.